blitzstudios/sleeper-mini
# Updating React Native to 72, and repack to 3.6.

### Description

This commit upgrades `react-native` from 0.66.5 to 0.72.6, and `@callstack/repack` from 2.7 to 3.6. It includes support for new architecture features (turbo modules and fabric), but they are not currently enabled and will require some tweaking if we decide to enable them in the future.

Most of this commit was done by following the upgrade steps from the official [upgrade helper](https://react-native-community.github.io/upgrade-helper/?from=0.66.5&to=0.72.6), but there were also many changes related to repack specifically, which is not an officially supported mode of running react native so it required some custom changes.

### How To Test

1. Connect to the dev version of sleeper running react native 72 and check that debugging and hot reloading from the mini is reflected on the main app as expected.
3. Run `yarn build-mini`, upload a mini built with react native 72, and verify it runs in production.
